cmd/diff.go

```diff
@@ -69,33 +70,60 @@ func checkFilenameFlag(_ []string) error {
 	return errors.New("Please include --types=file with the --filename flag")
 }
 
+func processImage(imageName string, imageMap map[string]*pkgutil.Image, wg *sync.WaitGroup, errChan chan<- error) {
```

Comment:
Is there a name for this function that might be a bit more specific to what this function is doing? I wasn't sure what `processImage` meant when I was reading this. 

It seems like the purpose of this function is to look up sometthing about the image (im not sure what tho - are we actually getting the bytes of the image itself, or looking up some metadata or something else?) and put that something into `imageMap`

So some other ideas:
- `AddImageToMap`
- `GetMetadataForImage`
- `FindImageSource`
- `DownloadImage`

(You might have some other name ideas tho since you probably know what exactly `getImageForName` is doing)

Reply:
This function is essentially just a concurrent wrapper around `getImageForName`, which retrieves a reference to an image and a reader for the actual bytes of the image. I can change this to `getImage` (the `imageMap` isn't really a useful struct here so I don't want to refer to it), but the reason I called it `processImage` is because `getImageForName` actually does a bit of image processing, i.e. unpacks the image filesystem to the local filesystem. I should probably just add some comments explaining what each function does :)